chore: add color documentation

bjohnsonrmn opened the pull request:
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [ ] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [ ] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.

---------
**Outline your feature or bug-fix below**

My code adds a color swatch component to the doc site to show the colors in our theme file. I also made a few edits to the `contributing.md` and I moved a few styles that are specific to the doc homepage so that they apply to that page only.

bjohnsonrmn commented on src/theme/GlobalStyles/Globals.styles.ts:
```diff
@@ -19,11 +19,6 @@ export const GlobalCSS = createGlobalStyle`
-    overflow-y: hidden;
-    overflow-x: hidden;
-    height: 100vh;
-    /* TODO: Change this back to #ffffff */
```
I moved this CSS to the index page because it seems to apply only there. Should I preserve this `TODO` comment, or does this change resolve it?

sharkparty replied:
Commented above. We should chat about a good strategy for the splash page. I'm definitely convinced my temp solution wasn't ideal, but I still don't know what the best temp solution is.

brandonlilly commented on CONTRIBUTING.md:
````diff
@@ -72,7 +72,7 @@ $ npm i
 
 ```sh
-$ yarn dev
+$ yarn storybook
````
Nice catch 👍 There's also the `npm run dev` command below that'll need to get updated

bjohnsonrmn replied:
✅ I'll update the `npm run dev`

brandonlilly commented on docs/src/components/ColorSwatchFamilyList/ColorSwatchFamilyList.component.tsx:
```diff
@@ -0,0 +1,82 @@
+          onClick={ () => setCopyChoice('hexCode')}
+        >
+        </input>
+        <label for="copyChoiceHexCode"> Hex Code</label>
```
React is giving a warning that the label doesn't work. It should be `htmlFor=`. But I prefer wrapping the element with the label over using `for`/`id`s anyway:
```html
        <label>
          <input type="radio" name="copyChoice" value="hexCode"
          /> Hex Code
        </label>
```
```jsx
        <label>
          <input type="radio" name="copyChoice" value="variableName"
          /> Variable Name
        </label>
```

sharkparty replied:
The wrapping syntax here I can go either way on. Just an FYI in a production app you'd still need the `htmlFor` as well as `aria` tags to promote a11y, but this is the docs site, so I'm not going to police that here. Further, I'm working on the `Input` for Anchor right now so eventually you'll get some of this for free.

bjohnsonrmn replied:
I'm going to look into the `htmlFor` and `aria` because I'd like to practice doing it the right way.

bjohnsonrmn replied:
Found this: https://a11y-style-guide.com/style-guide/section-forms.html#kssref-forms-radio-buttons

brandonlilly commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+// REACT
+import * as React from 'react';
+// VENDOR
+import styled, { css } from 'styled-components';
```
`{ css }` doesn't appear to be used

bjohnsonrmn replied:
✅ Will remove

brandonlilly commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+import { ColorContrastHelpers } from '../ColorContrastHelpers';
+import { ColorSwatch } from '../ColorSwatch';
+// THEME
+import { colors, sizes } from '../../../../src/theme';
```
`colors` also doesn't appear to be used

bjohnsonrmn replied:
I am using `colors` it on line 30:
`  box-shadow: 0px 0px 12px 4px ${colors.ash.light};`

But `sizes` looks like it isn't used 😂 so I will remove that.

brandonlilly commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+        textColor={textColor}
+        size={size}
+      >
+       <p class="ColorSwatchName">{name}</p>
```
`className`

sharkparty replied:
+1 on @brandonlilly's comment, but also: I built out a new component called Typography that can help do all the text formatting automatically. I'm going to dump this here with no explanation to see how effective Stories and IDE autocomplete is at telling a developer how to use it. If you hit turbulence using this, then blow me up immediately and that's what will inform what we document.

bjohnsonrmn replied:
@sharkparty the `Typography` component looks cool. Looking at the stories and the interface it seems intuitive to me how to use. I am running into 2 difficulties with it though.

1. I am setting the text color via a prop in the form of a hex code because it is dynamic based on color contrast. I have this prop available to pass in to `Typography` but it looks like its interface does not accept hex code values (also this seems like an edge case to me).

2. I'm wanting to zero out margin (like a normalize). Was that an intentional choice to leave that out?

I don't think either of these are worth making changes to `Typography` at the moment, but perhaps they are reasons for using classes for now, and I will use `Typography` for other simpler elements. What do you think?

brandonlilly commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+  textColor
+}: ColorSwatchProps): React.ReactElement<any> => {
+
+  let copyToClipboardText = backgroundColor;
```
just a suggestion, feel free to ignore
```js
const copyToClipboardText = copyChoice == 'variableName' ? backgroundColor : name
```

bjohnsonrmn replied:
I like it ✅

brandonlilly commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+// JSX MARKUP
+export const ColorSwatch = ({
+  className,
+  backgroundColor,
```
In my opinion, `ColorSwatch` should take a `color` or something similarly named (e.g. `swatchColor`, `themeColor` etc) as a prop to go along with the swatch `name`, which is then used as the background color, instead of specifying a backgroundColor explicitly. 

That way the way in which the swatch decides to present the swatch color is decoupled from its interface.

brandonlilly commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+    <ColorSwatch
+      className={classNames(className)}
+      backgroundColor={baseColorHexCode}
+      contrastOnBlack={ColorContrastHelpers.getRoundedColorContrast(baseColorHexCode, '#000000')}
```
I also think that the `contrastOnX` and `textColor` props should instead be calculated in the child `ColorSwatch`, rather than calculated here. All of them are just a function of the backgroundColor, so it wouldn't make sense to pass in any other values for a given color.

That way it's also much easier to change which colors we decide to contrast against in the future, and you aren't stuck doing this math every single time you want to use a `ColorSwatch`.

sharkparty replied:
+1

brandonlilly commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+}
+
+const StyledColorSwatch = styled.div`
+  background: ${props => props.backgroundColor };
```
We should add some indication that the ColorSwatches are clickable- ie. `cursor: pointer`.

Nice to haves:
 * A copy clipboard icon on hover either in the corner or next to the `name` or `color` that will be copied (depending on copyChoice).
 * Some small visual indication when text is copied

But don't let me feature creep you

sharkparty commented on docs/src/components/ColorContrastHelpers/ColorContrastHelpers.component.tsx:
```diff
@@ -0,0 +1,32 @@
+  }
+
+  public static getSuccessOrErrorColor(colorContrast): number {
+    if (colorContrast > 4.5) {
```
In cases where you're using a single predicate, you can use a ternary to abbreviate this a bit.
```
return 1 > 2 ? true : false;
```

bjohnsonrmn replied:
✅ Updated

sharkparty commented on docs/src/components/ColorContrastHelpers/ColorContrastHelpers.component.tsx:
```diff
@@ -0,0 +1,32 @@
+      return '#fff';
+    } else {
+      return '#000';
+    }
```
All of this can be:
```
return wcagContrast.hex('#fff', colorHexCode) > wcagContrast.hex('#000', colorHexCode) ? '#fff' : '#000';
```

zero0Halo replied:
I don't mind the verbosity. I got what was happening right away whereas the more terse line of code had me question my career choice.

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+const StyledColorSwatch = styled.div`
+  background: ${props => props.backgroundColor };
+  color: ${props => props.textColor };
+  padding: 24px 16px 16px 16px;
```
We can use `REM`s here (1rem = 16px) and we can abbreviate this CSS: `padding 1.5rem 1rem 1rem;`

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+// STYLES
+const ColorSwatchSizeStyles: ColorSwatchStylesGroup = {
+  large: css`
+    .ColorSwatchPillContainer {
```
All css classes should be kabob-case. `color-swatch-pill-container`

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+const ColorSwatchSizeStyles: ColorSwatchStylesGroup = {
+  large: css`
+    .ColorSwatchPillContainer {
+      padding-top: 200px
```
Why not use `height` to define a flex-box (FB) block element and then flex-end the content? Also, FB is new so if you haven't played with it, it's super cool.

sharkparty replied:
Also, always use `REM`'s plz.

bjohnsonrmn replied:
@sharkparty yeah I've only played with it a little, which is why I used a different way. I'd like to learn it, but I don't think I'd want to hold this ticket up for that unless you think it is important.

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+      >
+       <p class="ColorSwatchName">{name}</p>
+       <p class="ColorSwatchBackgroundColor">{backgroundColor}</p>
+       <div class="ColorSwatchPillContainer">
```
You may be able to use `Grid` and `Cell`s to do this more easily.

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+         <Pill value={contrastOnWhite} backgroundColor="#ffffff" textColor={backgroundColor}></Pill>
+         <Pill value={contrastOnFAFAFA} backgroundColor="#fafafa" textColor={backgroundColor}></Pill>
+         <Pill value={contrastOnF1F1F1} backgroundColor="#f1f1f1" textColor={backgroundColor}></Pill>
+         <Pill value={contrastOnBlack} backgroundColor="#000000" textColor={backgroundColor}></Pill>
```
```
[
    { key: 1, value: contrastOnWhite, backgroundColor: '#fff', textColor: backgroundColor },
    ...
].map((pillData: PillDataItem) =>(<Pill {...pillData}>));
```

bjohnsonrmn replied:
@sharkparty hmmm, I actually find that more difficult to read. Do you mind if I leave it?

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+
+// STYLES
+const StyledColorSwatchFamily = styled.div`
+  padding: 12px;
```
Same note on `REM`s

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+  margin: 24px;
+  width: 25%;
+  min-width: 300px;
+  display: inline-block;
```
Another shameless plug for flex-box.

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,87 @@
+    ></ColorSwatch>
+
+    { lightColorHexCode && (
+      <ColorSwatch
```
I think we can clean this up a little bit by separating the color swatch JSX from the logic determining what to render. [Google's implementation](https://github.com/mui-org/material-ui/blob/next/docs/src/pages/style/color/Color.js) for this is way different, but even at a high level look at how little JSX-foo is going on here.

bjohnsonrmn replied:
I'm about to push up some code with `ColorSwatch` refactored to have less props, so that should clean up this file a bit. Let me know if this still seems like an issue to you. Also, when I first started this ticket, some of the colors only had `base` defined, which is why I did that check. Perhaps we could remove if that colors file has a public contract to have `light` and `dark` also defined.

sharkparty commented on docs/src/components/ColorSwatchFamilyList/ColorSwatchFamilyList.component.tsx:
```diff
@@ -0,0 +1,82 @@
+
+// STYLES
+const StyledColorSwatchFamilyList = styled.div`
+
```
Doesn't look like we need this styled component.

bjohnsonrmn replied:
✅ I'll remove.

bjohnsonrmn replied:
I initially forgot to remove this, but I just removed it ✅

sharkparty commented on docs/src/components/ColorSwatchFamilyList/ColorSwatchFamilyList.component.tsx:
```diff
@@ -0,0 +1,82 @@
+      className={classNames(className)}
+    >
+
+      <p> Each color swatch has 4 pills representing the color contrast when used on these colors:</p>
```
If you import Typography this page will be a lot easier to deliver, unless I did a poor job building that component, in which case let's make it better!

sharkparty commented on docs/src/components/Pill/Pill.component.tsx:
```diff
@@ -0,0 +1,47 @@
+// STYLES
+export const StyledPill = styled.div`
+  color: ${props => props.textColor };
+  font-size: 16px;
```
Generic `REM`s comment

sharkparty commented on docs/src/pages/index.tsx:
```diff
@@ -1,12 +1,22 @@
 // THEME
 
+const IndexPageGlobalCSS = createGlobalStyle`
+  html,
```
This is a little alarming. What's this thing for?

bjohnsonrmn replied:
@sharkparty I moved those styles from `src/theme/GlobalStyles/Globals.styles.ts` to here because it seemed that they were necessary only on the index page (they seem to be for the wave animation).

sharkparty replied:
We should try to make them only impact this page and remove the global override, or leave them in there until we remove the splash page. There are definitely things in the globals just for the splash page, like that bright red BG, but I don't want to globalize that better, I want that entry to be refactored when we land the new index page.

sharkparty commented:
@bjohnsonrmn—great start here! This is really coming along nicely!

bjohnsonrmn commented:
@brandonlilly  and @sharkparty, thanks for all the good feedback. I'm planning to work through it today!

brandonlilly commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+       <p class="ColorSwatchBackgroundColor">{backgroundColor}</p>
+       <div class="ColorSwatchPillContainer">
+         <Pill value={contrastOnWhite} backgroundColor="#ffffff" textColor={backgroundColor}></Pill>
+         <Pill value={contrastOnFAFAFA} backgroundColor="#fafafa" textColor={backgroundColor}></Pill>
```
Minor suggestion: I suspect the pills would look more pleasing if they were a uniform width. Currently the width is being determined by however many digits the value happens to be.

bjohnsonrmn replied:
I like it. I updated to a uniform width ✅

bjohnsonrmn commented:
@sharkparty and @brandonlilly I just pushed fixes for most if not all the feedback, as noted in my comments, so it's ready for you to take another look at on Monday 👍

bjohnsonrmn commented:
@sharkparty and @brandonlilly, here are a couple things to note on the work I did for Galen's redesign:

- Made variable name, color hex code, and rgba code individually clickable, which removed the need for the toggle between variable name and hex code
- Renamed ColorContrastHelpers to ColorHelpers because I added a function that wasn't related to contrast
- Re-worked the HTML/CSS structure of the color swatch component even though the props remained largely the same

zero0Halo commented on docs/src/components/ColorContrastHelpers/ColorContrastHelpers.component.tsx:
```diff
@@ -0,0 +1,32 @@
+export class ColorContrastHelpers {
+
+  public static getRoundedColorContrast(colorHexCode1, colorHexCode2): number {
+    let exactContrast = wcagContrast.hex(colorHexCode1, colorHexCode2);
```
Is there a reason for **let** instead of **const** here?

bjohnsonrmn replied:
No. I re-wrote this to use no variable at all on Kameron's suggestion (just a return).

zero0Halo commented on docs/src/components/ColorContrastHelpers/ColorContrastHelpers.component.tsx:
```diff
@@ -0,0 +1,32 @@
+  }
+
+  public static getHigherContrastColor(colorHexCode): string {
+    let contrastOnBlack = wcagContrast.hex('#000', colorHexCode);
```
I've seen this in a few spots now, so I will only comment one more time on it. Consider using **const** instead of **let**. Also since you are reusing string values (i.e. '#000') it would probably be better to assign those to a const to avoid magic numbers.

bjohnsonrmn replied:
I think I was using `let` because I was changing the value later. I refactored this one as well to not have any variables. I did update to use named colors rather than hex codes.

zero0Halo commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,104 @@
+
+    <CopyToClipboard text={copyToClipboardText}>
+      <StyledColorSwatch
+        className={classNames(className)}
```
I might be missing it with the other comments breaking up the code, but is the usage of **classNames()** here necessary? There's no concatenation of classes that I can see, just the custom one the user can pass via props.

bjohnsonrmn replied:
I think I'm just using 1 class for the styled components, so I think you are right that I may be able to remove `classNames()`. Let me check!

bjohnsonrmn replied:
@zero0Halo this was a good catch. Not only was I able to remove `classNames()`, I removed the `className` prop as well! I think I thought that it was necessary for styled components somehow, but it seems it isn't according to these docs: https://www.styled-components.com/docs/basics#how-do-styled-components-work-within-a-component

zero0Halo commented on docs/src/components/ColorSwatchFamilyList/ColorSwatchFamilyList.component.tsx:
```diff
@@ -0,0 +1,82 @@
+          id="copyChoiceVariableName"
+          name="copyChoice"
+          value="variableName"
+          checked={copyChoice == 'variableName'}
```
I've noticed it in a few spots but I'll mention it here. Generally best to use === instead of == for comparisons. This checks for type as well.

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,105 @@
+  }
+
+  .temporary-copy-confirmation {
+    font-size: 1rem;
```
All of this can come from the Typography component that probably didn't exist when you started this work. @bjohnsonrmn I know you've got other things on your plate and this should have been called out earlier, so would you like me to refactor this with Typography, or would you like to work on that?

bjohnsonrmn replied:
@sharkparty I used `Typography` for some other elements, but this one seemed complex enough to deserve a class. I can re-work it, but are you saying every CSS attribute can be worked into `Typography` or only some of them?

sharkparty replied:
The only ones that can't would be position, top, right, border, & bg. A few of those will be changing when we build out a composable base component. I can definitely be convinced this is better. Are those the only props and usage that's outside of Typo...y?

bjohnsonrmn replied:
We talked this over in person and decided to leave as is 👍

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,105 @@
+  const highContrastBorderColor = ColorHelpers.getHigherContrastColor(colorHexCode);
+  const colorRgbCode = ColorHelpers.getRgbColor(colorHexCode)
+
+  const [isCopied, setIsCopied] = useState(false);
```
Did you know you can type your hooks? Me neither until I did some digging:

```
useState<boolean>(false);
```

bjohnsonrmn replied:
I didn't, but that's cool. I'll update!

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,105 @@
+
+  const [isCopied, setIsCopied] = useState(false);
+  const [copyToClipboardText, setCopyToClipboardText] = useState(colorHexCode);
+  const handleCopy = function(copiedText) {
```
Do you need `function`-scope? or can this be an arrow function?

bjohnsonrmn replied:
I think this could be an arrow function 👍

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,105 @@
+      colorHexCode={colorHexCode}
+      highContrastBorderColor={highContrastBorderColor}
+    >
+     {isCopied ? <span className="temporary-copy-confirmation">🎉 You copied "{copyToClipboardText}"</span> : null}
```
First emoji in the codebase!

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,105 @@
+
+     <CopyToClipboard text={colorHexCode} onCopy={() => handleCopy(colorHexCode)}>
+       <div className="color-swatch-pill-container">
+         <Pill value={ColorHelpers.getRoundedColorContrast(colorHexCode, '#ffffff')} backgroundColor="#ffffff" textColor={colorHexCode}></Pill>
```
Could this be:

```
{['#ffffff', '#fafafa', ...].map(() => (
    <Pill
        key={color}
        value={ColorHelpers.getRoundedColorContrast(colorHexCode, `${color}`)}
        backgroundColor={color}
        textColor={colorHexCode}
    />
))}
```

bjohnsonrmn replied:
Yes! Will do.

sharkparty commented on docs/src/components/Pill/Pill.component.tsx:
```diff
@@ -0,0 +1,45 @@
+
+// STYLES
+export const StyledPill = styled.div`
+  color: ${props => props.textColor };
```
We'll wanna set the `box-sizing` of this to `border-box` ([sauce](https://developer.mozilla.org/en-US/docs/Web/CSS/box-sizing)) to make sure all your padding and margins are predictably rendered without exploding a DOM block.

bjohnsonrmn replied:
Thanks for this catch 🐟

sharkparty commented on docs/src/components/ColorHelpers/ColorHelpers.component.tsx:
```diff
@@ -0,0 +1,31 @@
+    let h=colorHexCode.replace('#', '');
+    h =  h.match(new RegExp('(.{'+h.length/3+'})', 'g'));
+
+    for(var i=0; i<h.length; i++)
```
`for(let i=0; i<h.length; i++)`

sharkparty commented on docs/src/components/ColorHelpers/ColorHelpers.component.tsx:
```diff
@@ -0,0 +1,31 @@
+    for(var i=0; i<h.length; i++)
+        h[i] = parseInt(h[i].length===1? h[i]+h[i]:h[i], 16);
+
+    return 'rgb('+h.join(',')+')';
```
```
return `rgb(${h.join(',')})`;
```

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,110 @@
+    padding: 7rem 1rem 1rem 1.5rem;
+  }
+
+  .color-swatch-text-container {
```
color-swatch-text-container can be its own styled component to house all these styles more neatly.

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,110 @@
+    padding: 1rem;
+  }
+
+  .color-swatch-text-container p {
```
This nested rule can be inside the curly brackets of the container:
```
color-swatch-text-container p {}
```
```
color-swatch-text-container {
      p { }
}
```

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,110 @@
+    padding: 0.25rem 0 0.25rem 0.5rem;
+  }
+
+  .color-swatch-text-container p:hover {
```
This can be nested very much the same way:

```
.color-swatch-text-container p:hover {}
```
can be
```
.color-swatch-text-container {
    p {
        &:hover {}
    }
}
```

sharkparty replied:
Just remember:
> Note that objects should have at most 4 levels. Most of the time you'll stay around the 2-3 level range, and the fourth would be the interaction state.
~[Sauce](http://thesassway.com/beginner/the-inception-rule)

bjohnsonrmn replied:
Will do. I've tasted this sauce before. It takes me back to Tuscany. 🍷

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,110 @@
+    background: ${colors.ash.light};
+    border-radius: 1rem;
+    cursor: pointer;
+    text-decoration: underline;
```
```
...
cursor: pointer;
text-decoration: underline;
...
```

If you're writing all of these styles that are defaults for the anchor tag, wouldn't it be better to use the anchor tag?

bjohnsonrmn replied:
This is a good question. In this case, I would rather add a few anchor-tag-like qualities to `<p>` tags than use `<a>` and have to subtract a few anchor-tag-like qualities I don't want (blue color, inline display, etc.).

sharkparty commented on docs/src/components/ColorSwatch/ColorSwatch.component.tsx:
```diff
@@ -0,0 +1,110 @@
+        </div>
+     </CopyToClipboard>
+     <div className="color-swatch-text-container">
+       <CopyToClipboard text={colorName} onCopy={() => handleCopy(colorName)}><Typography as="p" weight="700">{colorName}</Typography></CopyToClipboard>
```
This is now
```
<Typography as="p" weight="700">
```
this
```
<Typography tag="p" weight="700">
```
because of a bug identified by @brandonlilly

bjohnsonrmn replied:
It looks like `Typography` was also updated to zero out margin and padding, which I was taking advantage of before. What is the preferred way to add in margin or padding?

bjohnsonrmn replied:
✅ I added this margin back in using the styled component.

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,57 @@
+
+// STYLES
+const StyledColorSwatchFamily = styled.div`
+  margin: 1rem 0;
```
`box-sizing: border-box;`

bjohnsonrmn replied:
How are you determining which components need this?

sharkparty replied:
They all do, but containing components or anything that has padding/margins added usually catch my eye.

bjohnsonrmn replied:
@sharkparty sounds good. Where's the sauce?

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,57 @@
+
+    <ColorSwatch
+      colorHexCode={baseColorHexCode}
+      colorName={baseColorName+'.base'}
```
```
colorName={`${baseColorName}.base`}
```

sharkparty replied:
[sauce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

bjohnsonrmn replied:
The sauce was a fascinating read / sumptuous flavor journey 🍝

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,57 @@
+    { lightColorHexCode && (
+      <ColorSwatch
+        colorHexCode={lightColorHexCode}
+        colorName={baseColorName+'.light'}
```
String template literal

sharkparty commented on docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx:
```diff
@@ -0,0 +1,57 @@
+    { darkColorHexCode && (
+      <ColorSwatch
+        colorHexCode={darkColorHexCode}
+        colorName={baseColorName+'.dark'}
```
String template literal

sharkparty commented on docs/src/pages/index.tsx:
```diff
@@ -7,7 +7,11 @@ import { withPrefix } from 'gatsby';
 const StyledIndexPage = styled.div`
+    overflow-y: hidden;
+    overflow-x: hidden;
+    height: 100vh;
```
👍

bjohnsonrmn commented:
@sharkparty I rebased and addressed all but two of your comments, on which I left questions for you.
